pkg/event/controller/controller_test.go

```diff
@@ -0,0 +1,160 @@
+package controller_test
+
+import (
+	"fmt"
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/admiral/pkg/fake"
+	"github.com/submariner-io/admiral/pkg/syncer/test"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/event/controller"
+	"github.com/submariner-io/submariner/pkg/event/testing"
+)
+
+const testNamespace = "test-namespace"
+const testHandlerName = "test-handler"
+const testLocalClusterID = "local-cluster"
+
+var _ = Describe("Event controller", func() {
+	var (
+		endpoints  dynamic.ResourceInterface
+		nodes      dynamic.ResourceInterface
+		node       *corev1.Node
+		endpoint   *submV1.Endpoint
+		hostname   string
+		testEvents chan testing.TestEvent
+		stopCh     chan struct{}
+		registry   event.Registry
+	)
+
+	BeforeEach(func() {
+		stopCh = make(chan struct{})
+		testEvents = make(chan testing.TestEvent, 1000)
+		testHandler := testing.NewTestHandler(testHandlerName, event.AnyNetworkPlugin, testEvents)
+		registry = event.NewRegistry("test-registry", "test-plugin")
+		_ = registry.AddHandlers(testHandler)
+		hostname, _ = os.Hostname()
+		node = NewNode(hostname)
+
+	})
+
+	JustBeforeEach(func() {
+		_ = submV1.AddToScheme(scheme.Scheme)
+
+		config := controller.Config{
+			RestMapper: test.GetRESTMapperFor(&corev1.Node{}, &submV1.Endpoint{}),
+			Client:     fake.NewDynamicClient(scheme.Scheme),
+			Registry:   &registry,
+		}
+		os.Setenv("SUBMARINER_NAMESPACE", testNamespace)
+		os.Setenv("SUBMARINER_CLUSTERID", testLocalClusterID)
+
+		nodes = config.Client.Resource(*test.GetGroupVersionResourceFor(config.RestMapper, &corev1.Node{}))
+		endpoints = config.Client.Resource(*test.GetGroupVersionResourceFor(config.RestMapper,
+			&submV1.Endpoint{})).Namespace(testNamespace)
+
+		eventController, err := controller.New(&config)
+
+		Expect(err).To(Succeed())
+		go func() {
+			Expect(eventController.Start(stopCh)).To(Succeed())
+		}()
+	})
+
+	AfterEach(func() {
+		close(stopCh)
+	})
+
+	When("a Node is created, updated and deleted", func() {
+		It("should notify the appropriate handler of each event", func() {
+			obj := test.CreateResource(nodes, node)
+			node.Namespace = obj.GetNamespace()
+			node.ResourceVersion = obj.GetResourceVersion()
+			node.UID = obj.GetUID()
+
+			Eventually(testEvents).Should(Receive(Equal(
+				testing.TestEvent{Handler: testHandlerName, Name: testing.EvNodeCreated, Parameter: node})))
+			Consistently(testEvents).ShouldNot(Receive())
+
+			node.Labels = map[string]string{"labeled-i-am": "i-am"}
+
+			test.UpdateResource(nodes, node)
+
+			Eventually(testEvents).Should(Receive(Equal(
```

Comment:
@mangelajo cool. So you added a test-case to verify that we are not getting events when labels are updated. Nice.

Reply:
I actually removed that filter for now. I will be needing labels and annotations for OVN.

We can later add some interface from the controller consumers to let us install specific filters. I hope that sounds ok @sridhargaddam

Reply:
Okay @mangelajo